.github/workflows/internal-images.yml

```
@@ -0,0 +1,46 @@
+# yamllint --format github .github/workflows/internal-images.yml
+---
+name: internal-images
+
+# Refresh the tags once a day
+on:
+  schedule:
+    - cron: "23 3 * * *"
+  workflow_dispatch:  # Allows manual refresh
+
+# This copies images from docker.io to ghcr.io/tetratelabs/getenvoy-internal:$tag
+# Using these in tests and as a parent (FROM) avoids docker.io rate-limits particularly on pull requests.
+jobs:
+  copy-images:
+    strategy:
+      matrix:
+        # Be precise in tag versions to improve reproducibility
+        include:
+          - source: busybox:1.32.1  # test image: ci/e2e/darwin/install_docker.sh
+            target_tag: busybox
+          - source: registry:2  # test image: docker-compose.yml
+            target_tag: registry
+          - source: rust:1.51.0  # parent image: images/extension-builders/rust/Dockerfile
+            target_tag: rust
+          - source: tinygo/tinygo:0.17.0  # parent image: images/extension-builders/tinygo/Dockerfile
+            target_tag: tinygo
```

I would prefer to include the version info in the docker hub. What if the container updated while building the release containers? That would change the assumption that the container built at release job would pass all the unittest.. WDYT?

```suggestion
        include:
          - source: busybox:1.32.1  # test image: ci/e2e/darwin/install_docker.sh
            target_tag: busybox-1.32.1
          - source: registry:2  # test image: docker-compose.yml
            target_tag: registry-2
          - source: rust:1.51.0  # parent image: images/extension-builders/rust/Dockerfile
            target_tag: rust-1.15.0
          - source: tinygo/tinygo:0.17.0  # parent image: images/extension-builders/tinygo/Dockerfile
            target_tag: tinygo-0.17.0
```

---

So, this topic effects the builder images as the test ones don't matter.

Changes to this file apply on the cron, which is 3:23 am UTC. So, if someone updated, it wouldn't happen immediately. The risk would be is the release is going on and passes 3:23am UTC where this file was updated. I think this is a low risk because releases don't occur that frequently.

If we solved like this, we would introduce a double-maintenance problem of fixing the image version here, waiting for it to be published, then changing in the corresponding files. Almost certainly, we would end up with drift because someone will forget to update twice.

Meanwhile, we currently only push "latest" tags. What this means is even if we prevent simultaneous updates, the next time the builder image is pushed, any version will get that whether they like it or not.

We can eventually pin releases to a docker tag, but while we don't I think the approach here doesn't make as much sense as it seems to at first glance, so leaving it alone is probably best. wdyt?